lib/model/day_tracked_data.dart

```diff
@@ -14,6 +14,32 @@ class DayTrackedData {
     });
     return str;
   }
+
+  bool hasComment() {
+    return entries.containsKey(TrackedDataKeys.comment);
+  }
+
+  bool hasDidPlayFavorite() {
+    return entries.containsKey(TrackedDataKeys.didPlayFavorite);
+  }
+
+  String getComment() {
+    final trackedDataValue = entries[TrackedDataKeys.comment];
+    if (trackedDataValue != null &&
+        trackedDataValue.runtimeType == TextTrackedValue)
+      return (trackedDataValue as TextTrackedValue).text;
```

Comment:
Do we expect incorrect data types here? Couldn't the whole method be simplified to something like 

```suggestion
      return (entries[TrackedDataKeys.comment] as TextTrackedValue)?.text;
```

Reply:
I mean it's a security, I would say there is never too much security.

Reply:
Idk, for me this is code noise. I'd much rather write unit tests to make sure this never happens.

Also, this will hide errors. If we have some errors which allow this to happen I'd rather we discover it via a crash and fix it, as opposed to silently displaying no records.

Reply:
So the enterprise "banking-app" way to write this would be your way but also with added logging/reporting.

I think for our case we can just go the straightforward way though.